### DISCUSSION TITLE
About pinning plugins commit

### DISCUSSION THREAD

**Discussion by xeluxee** [LunarVim/LunarVim#1995] [General]
I agree with pinning some plugins which often introduce breaking changes, such as NvimTree
But it's worth it to pin every single core plugin? That causes some annoyances, especially when there are important updates: users have to make a PR to update commit or wait for the pin to be updated. For example `Comment.nvim` received an important update recently; it introduced integration with treesitter for commenting, but I can't update it due to the pin.

Also having 30~40 core plugins makes it difficult to keep everything updated.

At least it'd be better to make commits customizable, using something like `lvim.plugins_commit`

**Comment by chase** [comment]
For the stable branch, it is very important. It's also impossible to tell when a breaking change will be introduced. For example barbar.nvim one day suddenly started crashing lvim because a PR was merged without being tested.

It's a lot of fires to put out, but you're absolutely right, on my own branch I've been keeping nvim-cmp up to date because of some nice behavior improvements but I haven't pushed for updates in LunarVim because it's tedious to get approval.

What we could do is export `commit` from here: `https://github.com/LunarVim/LunarVim/blob/rolling/lua/lvim/plugins.lua#L1`
`nvim-cmp`'s commit hash would be accessible by `require('lvim.plugins').commit.nvim_cmp` for example. What do you think @xeluxee?

**Reply by xeluxee** [reply]
> What we could do is export `commit` from here: https://github.com/LunarVim/LunarVim/blob/rolling/lua/lvim/plugins.lua#L1

I think it's the best thing to do. What about exporting it to `lvim.core_plugins_commit`?
So changing hash would be simple as adding this to `config.lua`:
``` lua
lvim.core_plugins_commit.nvim_cmp = "hash"
```

**Reply by kylo252** [reply]
I have a plan to make them updated regularly via a github action. However, I've been struggling with Packer which is currently broken/unreliable in headless mode.  That's what I'm trying to fix currently.

**Reply by xeluxee** [reply]
What does "regularly" mean? Every week? Or every month?
Anyway why not leaving user the possibility to customize plugin hash using something like `lvim.core_plugins_commit`?

**Reply by kylo252** [reply]
For rolling, I was thinking it's probably daily. I'm not sure about master, since chris has decided to freeze it until 1.0.

Regarding the second part, there was an attempted PR for this at some point but the author closed it, and said he'd come back with a more polished solution iirc.

**Comment by chase** [comment]
Also related is @kylo252's work here: https://github.com/LunarVim/LunarVim/pull/1989
